### PR TITLE
feat: add admin search query oversight endpoint

### PR DESCRIPTION
**Pull request by swapnil-turing**
# Fixes Issue

**My PR closes #issue_number_here**

# 👨‍💻 Changes proposed(What did you do ?)

# ✔️ Check List (Check all the applicable boxes)

- [] My code follows the code style of this project.
- [] This PR does not contain plagiarized content.
- [] The title and description of the PR is clear and explains the approach.

##  Note to reviewers

# 📷 Screenshots

### CODE REVIEW

File: augment-store/server/products/views.py [hunk 1]
```diff
@@ -275,3 +279,13 @@ def retrieve(self, request, *args, **kwargs):
         serializer = self.get_serializer(data)
         return Response(serializer.data)
 
+
+class AdminSearchQueryListView(CachedListMixin, AutoOptimizeMixin, ListAPIView):
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/products/views.py:283]: value:delightful; category:bug; feedback:caught a cache collision vulnerability caused by missing cache namespace isolation. Rated 'delightful' because shared cache keys silently serve stale or wrong data across endpoints.

File: augment-store/server/products/views.py [hunk 2]
```diff
@@ -275,3 +279,13 @@ def retrieve(self, request, *args, **kwargs):
...
+    """Admin-only view to list all search queries globally."""
+    serializer_class = SearchQueryListSerializer
+    permission_classes = [IsAuthenticated, hasAdminRole]
+    queryset = SearchQuery.objects.all()
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/products/views.py]: value:delightful; category:bug; feedback:caught an unstable pagination bug caused by missing explicit ordering on the queryset. Rated 'delightful' because this leads to duplicate or omitted rows across pages.

File: augment-store/server/products/urls_products.py [hunk 3]
```diff
@@ -18,4 +19,5 @@
     path('featured/', FeaturedProductListView.as_view(), name='product_featured'),
     path('recommend/', RecommendProductListView.as_view(), name='product_recommend'),
     path('admin/<uuid:pk>/', AdminProductUpdateDeleteView.as_view(), name='admin_product_update_delete'),
+    path('admin/search-queries/', AdminSearchQueryListView.as_view(), name='admin_search_query_list'),
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/products/urls_products.py:22]: value:useful; category:documentation; feedback:caught incomplete PR metadata. Rated 'useful' because polished PR descriptions aid in readability and tracing.

**Comment by swapnil-turing** [comment]
augment review

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/products/tests.py [hunk 4]
```diff
@@ -1728,3 +1728,34 @@ def test_admin_delete_product(self):
...
+        self.admin_query = SearchQuery.objects.create(query="laptop", results_count=10, user=self.admin_user)
+        self.regular_query = SearchQuery.objects.create(query="phone", results_count=5, user=self.regular_user)
+        
+        self.url = reverse('v1:admin_search_query_list')
+
+    def test_admin_can_list_search_queries(self):
+        self.client.force_authenticate(user=self.admin_user)
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        results = response.data.get('results', [])
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/products/tests.py]: value:delightful; category:bug; feedback:caught a response shape assumption that would raise AttributeError when pagination is disabled. Rated 'delightful' because consistency with existing test patterns prevents silent failures.

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/products/serializers.py [hunk 5]
```diff
@@ -93,3 +93,9 @@ class ProductDetailSerializer(serializers.ModelSerializer):
     class Meta:
         model = Product
         fields = "__all__"
+
+class SearchQueryListSerializer(serializers.ModelSerializer):
+    user = UserListSerializer(read_only=True)
+    class Meta:
+        model = SearchQuery
+        fields = "__all__"
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/products/serializers.py]: value:delightful; category:security; feedback:caught an over-exposure risk from using __all__ in a serializer. Rated 'delightful' because explicit field lists prevent accidental data leakage when models evolve.

File: augment-store/server/products/views.py [hunk 6]
```diff
@@ -275,3 +279,12 @@ def retrieve(self, request, *args, **kwargs):
         serializer = self.get_serializer(data)
         return Response(serializer.data)
 
+
+class AdminSearchQueryListView(CachedListMixin, AutoOptimizeMixin, ListAPIView):
+    """Admin-only view to list all search queries globally."""
+    serializer_class = SearchQueryListSerializer
+    permission_classes = [IsAuthenticated, hasAdminRole]
+    cache_service_class = SearchQueryCacheService
+    queryset = SearchQuery.objects.all().order_by('-created_at', '-id')
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/products/views.py]: value:delightful; category:bug; feedback:caught soft-deleted rows leaking into the admin listing. Rated 'delightful' because consistent soft-delete filtering is critical for data integrity.

File: augment-store/server/products/views.py [hunk 7]
```diff
@@ -275,3 +279,12 @@ def retrieve(self, request, *args, **kwargs):
         serializer = self.get_serializer(data)
         return Response(serializer.data)
 
+
+class AdminSearchQueryListView(CachedListMixin, AutoOptimizeMixin, ListAPIView):
+    """Admin-only view to list all search queries globally."""
+    serializer_class = SearchQueryListSerializer
+    permission_classes = [IsAuthenticated, hasAdminRole]
+    cache_service_class = SearchQueryCacheService
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/products/views.py:287]: value:delightful; category:bug; feedback:caught a cache keying mismatch where a global view was being cached per-user. Rated 'delightful' because this causes unnecessary cache duplication and potential staleness across admin users.

**Comment by swapnil-turing** [comment]
augment review

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/products/tests.py [hunk 8]
```diff
@@ -1728,3 +1728,34 @@ def test_admin_delete_product(self):
         self.assertEqual(list_response.status_code, 200)
         results = list_response.data.get('results', list_response.data) if isinstance(list_response.data, dict) else list_response.data
         self.assertEqual(len(results), 0)
+
+
+class AdminSearchQueryTests(BaseAPITestCase):
+    def setUp(self):
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/products/tests.py:1734]: value:delightful; category:bug; feedback:caught a test isolation issue where stale global cache could leak between test cases. Rated 'delightful' because cache-aware test setup is essential for reliable test suites.

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/products/views.py [hunk 9]
```diff
@@ -275,3 +279,19 @@ def retrieve(self, request, *args, **kwargs):
         serializer = self.get_serializer(data)
         return Response(serializer.data)
 
+
+class AdminSearchQueryListView(CachedListMixin, AutoOptimizeMixin, ListAPIView):
+    """Admin-only view to list all search queries globally."""
+    serializer_class = SearchQueryListSerializer
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/products/views.py:285]: value:delightful; category:bug; feedback:caught an N+1 query vulnerability from missing auto_select_related on a nested serializer relation. Rated 'delightful' because this directly impacts endpoint performance at scale.

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/products/tests.py [hunk 10]
```diff
@@ -1728,3 +1728,36 @@ def test_admin_delete_product(self):
...
+        super().setUp()
+        from products.services import SearchQueryCacheService
+        SearchQueryCacheService().clear_namespace()
+        from accounts.factory import UserFactory
+        self.admin_user = UserFactory(role='admin')
+        self.regular_user = UserFactory(role='member')
+        
+        self.admin_query = SearchQuery.objects.create(query="laptop", results_count=10, user=self.admin_user)
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/products/tests.py:1742]: value:delightful; category:bug; feedback:caught a missing edge-case test for nullable FK serialization. Rated 'delightful' because null-user rows are a real production scenario that can cause serializer errors if not covered.

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/products/views.py [hunk 11]
```diff
@@ -275,3 +279,20 @@ def retrieve(self, request, *args, **kwargs):
...
+    permission_classes = [IsAuthenticated, hasAdminRole]
+    cache_service_class = SearchQueryCacheService
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/products/views.py:287]: value:delightful; category:bug; feedback:caught a stale cache issue where the write-path didn't invalidate the read-path cache. Rated 'delightful' because cache coherence between write and read paths is essential for real-time admin oversight.

File: augment-store/server/products/services.py [hunk 12]
```diff
@@ -89,6 +89,7 @@ def log_search(query_string: str, results_count: int, user: User = None):
                 results_count=results_count,
                 user=user if user and user.is_authenticated else None
             )
+            SearchQueryCacheService().clear_namespace()
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/products/services.py]: value:delightful; category:bug; feedback:caught an error-handling conflation where cache failures could obscure successful DB writes. Rated 'delightful' because accurate error reporting is essential for debugging production issues.

**Comment by swapnil-turing** [comment]
augment review

**Review by michaelos443 — APPROVED** [review]
